test/extended/router/router.go

```diff
@@ -30,16 +29,24 @@ var _ = g.Describe("[Conformance][Area:Networking][Feature:Router]", func() {
 	// hook
 	g.AfterEach(func() {
 		if g.CurrentGinkgoTestDescription().Failed {
-			exutil.DumpPodLogsStartingWithInNamespace("router", "default", oc.AsAdmin())
-			selector, err := labels.Parse("router=router")
-			if err != nil {
-				panic(err)
+			currlabel := "router=router"
+			for _, ns := range []string{"default", "openshift-ingress", "tectonic-ingress"} {
```

Comment:
There's already a [GetRouterPodTemplate](https://github.com/openshift/origin/blob/master/test/extended/util/framework.go#L1444) (which itself needs updated to locate the new daemonset); couldn't there also be a util to get all the pods selected by the deployment/daemonset?

Reply:
I did a lot of refactoring of that function and concluded it was simpler not tu use it since it checks deployments and deploymentconfigs and it here it was only used for getting the namespace.
I want to address this function refactoring in another PR because it's used in other places of the test suite.

Reply:
I wasn't suggesting using it here, I was saying:

1. GetRouterPodTemplate is still used by tests and doesn't look like it would work for 4.0 clusters with our routers (as it's not looking for daemonsets)
2. A new util *similar to* GetRouterPodTemplate could. be created to find endpoints

Reply:
oh ok, i'll checko out how refactor or substitute this function